src/engine/strat_engine/thinpool/thinpool.rs

```diff
@@ -935,6 +935,50 @@ mod tests {
         assert!(pool.get_filesystem_by_uuid(fs_uuid).is_none());
     }
 
+    #[test]
+    pub fn loop_test_meta_expand() {
+        // This test requires more than 1 GiB.
+        loopbacked::test_with_spec(loopbacked::DeviceLimits::Range(2, 3), test_meta_expand);
+    }
+
+    #[test]
+    pub fn real_test_meta_expand() {
+        real::test_with_spec(real::DeviceLimits::Range(2, 3), test_meta_expand);
+    }
+
+    fn test_meta_expand(paths: &[&Path]) -> () {
+        let pool_uuid = Uuid::new_v4();
+        let dm = DM::new().unwrap();
+        let mut mgr = BlockDevMgr::initialize(pool_uuid, paths, MIN_MDA_SECTORS, false).unwrap();
+        // Create a ThinPool with a very small meta device.
+        let mut thin_pool = ThinPool::new(pool_uuid,
+                                          &dm,
+                                          DATA_BLOCK_SIZE,
+                                          Some(MetaBlocks(16)),
+                                          DATA_LOWATER,
+                                          &mut mgr)
+                .unwrap();
+
+        match thin_pool.status(&dm).unwrap() {
+            dm::ThinPoolStatus::Working(ref status) => {
+                let usage = &status.usage;
+                assert_eq!(usage.total_meta.0, MetaBlocks(16).0);
```

Comment:
```assert_eq!(usage.total_meta, MetaBlock(16))``` ought to work since both are meta blocks.

Reply:
ok